File: examples/js/src/001-translation.ts
import fs from "fs/promises";
import { ChatAnthropic } from "@langchain/anthropic";

const model = new ChatAnthropic({
  model: "claude-opus-4-8",
});

async function translate(text: string, language: string): Promise<string> {
  const response = await model.invoke([
    {
      role: "system",
      content: `You are a helpful assistant that translates text from English to ${language}.
      Translate the given text to ${language}. Return only the translated text, no other text.`,
    },
    {
      role: "user",
      content: text,
    },
  ]);
  return response.content.toString();
}

async function updateTranslations() {
  const translations = {
    en: JSON.parse(await fs.readFile("public/locales/en.json", "utf8")),
    de: JSON.parse(await fs.readFile("public/locales/de.json", "utf8")),
  };

  await Promise.all(
    Object.keys(translations.en).map(async (key) => {
      const enTranslation = translations.en[key];
      if (!translations.de[key]) {
        translations.de[key] = await translate(enTranslation, "de");
      }
    }),
  );

  console.log(translations);
  // TODO: implement write to file
}

updateTranslations();

File: examples/python/src/001_translation.py
import json
import asyncio
from langchain_anthropic import ChatAnthropic

model = ChatAnthropic(model="claude-opus-4-8")


async def translate(text: str, language: str) -> str:
    response = await model.ainvoke([
        {
            "role": "system",
            "content": f"You are a helpful assistant that translates text from English to {language}.\n"
                       f"Translate the given text to {language}. Return only the translated text, no other text.",
        },
        {
            "role": "user",
            "content": text,
        },
    ])
    return response.content


async def update_translations():
    with open("locales/en.json", "r") as f:
        en = json.load(f)
    with open("locales/de.json", "r") as f:
        de = json.load(f)

    tasks = []
    keys_to_translate = []
    for key in en:
        if key not in de:
            keys_to_translate.append(key)
            tasks.append(translate(en[key], "de"))

    results = await asyncio.gather(*tasks)
    for key, result in zip(keys_to_translate, results):
        de[key] = result

    print(json.dumps({"en": en, "de": de}, indent=4))
    # TODO: implement write to file


asyncio.run(update_translations())
